generate_favicons.sh

```
@@ -0,0 +1,42 @@
+then
+    echo "ImageMagick not found. Installing..."
+    apt-get update && apt-get install -y imagemagick
+    echo "ImageMagick installed."
+else
+    echo "ImageMagick is already installed."
+fi
+
+# Navigate to the public directory
+# Create the directory if it doesn't exist
+mkdir -p client/public
+cd client/public
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)